opm/simulators/wells/BlackoilWellModel.hpp

```diff
@@ -398,6 +398,7 @@ class WellContributions;
             Scalar gravity_{};
             std::vector<Scalar> depth_{};
             bool alternative_well_rate_init_{};
+            std::optional<double> well_group_thp_calc_;
```

Comment:
I see it is single value for the `BlackoilWellModel`. It might not work for multiple auto-choke nodes. And it might be wrong for parallel running, where each process has its own `well_group_thp_calc_`. 

Should we get this value from GroupState::group_thp?

Reply:
I have tried to use `GroupState::group_thp` as we have discussed but that I could not get to work. Your solution to  use `std::map<std::string, Scalar> well_group_thp_calc_` in `BlackoilWellModel.hpp` is working so I followed that. Many thanks for this!